src/components/rangeselector/defaults.js

```diff
@@ -0,0 +1,67 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+
+var attributes = require('./attributes');
+var buttonAttrs = require('./button_attributes');
+
+
+module.exports = function rangeSelectorDefaults(containerIn, containerOut, layout) {
+    var selectorIn = containerIn.rangeselector || {},
+        selectorOut = containerOut.rangeselector = {};
+
+    function coerce(attr, dflt) {
+        return Lib.coerce(selectorIn, selectorOut, attributes, attr, dflt);
+    }
+
+    var buttons = buttonsDefaults(selectorIn, selectorOut);
+
+    var visible = coerce('visible', buttons.length > 0);
```

Comment:
At the moment, range selector don't ship with any default buttons.

Shipping default buttons would require adding some logic based on the range of the time series, which wouldn't be too hard, but might be too *magical*.

Thoughts?